src/module/actor/mixins/actor-conditions.js

```diff
@@ -66,14 +111,51 @@ export const ActorConditionsMixin = (superclass) => class extends superclass {
             if (conditionItem) {
                 const promise = this.deleteEmbeddedDocuments("Item", [conditionItem.id]);
                 promise.then(() => {
-                    const updateData = {};
-                    updateData[`data.conditions.${conditionName}`] = false;
-                    this.update(updateData).then(() => {
+                    this._updateActor(conditionName, false).then(() => {
                         Hooks.callAll("onActorSetCondition", {actor: this, item: conditionItem, conditionName: conditionName, enabled: enabled});
                     });
                 });
+
                 return promise;
             }
         }
     }
+
+    /**
+     * Updates the actor data with the condition settings and then checks if Flat-Footed needs to be updated.
+     *
+     * @param {String} conditionName The name of the condition matching keys from config.js SFRPG.statusEffects
+     * @param {Boolean} enabled If this value is true it enables the condition
+     * @private
+     * */
+    async _updateActor(conditionName, enabled) {
+        const updateData = {};
+        updateData[`data.conditions.${conditionName}`] = enabled;
+
+        this.update(updateData).then(() => {
+            this._checkFlatFooted(conditionName, enabled);
+        });
+    }
+
+    /**
+     * Checks if the Flat-Footed condition should be enabled based on other conditions in SFRPG.conditionsCausingFlatFooted.
+     * @param conditionName The condition being added or removed
+     * @param enabled True if the condition is being added
+     * @private
+     */
+    _checkFlatFooted(conditionName, enabled) {
```

Comment:
This should be marked async as well, and return this.setCondition .., promise should always be propagated up.

This way we can also ensure the onActorSetCondition callback is called after flat footed is potentially also applied.